packages/babel-plugin-transform-block-scoping/src/validation.ts

```diff
@@ -197,7 +197,7 @@ function injectTDZChecks(binding: Scope.Binding, state: PluginPass) {
   for (const path of binding.referencePaths as NodePath<t.Identifier>[]) {
     if (path.parentPath.isUpdateExpression()) continue;
     // It will be handled after transforming the loop
-    if (path.parentPath.isFor({ left: path.node })) continue;
+    if (path.parentPath.isForXStatement({ left: path.node })) continue;
```

Comment:
This is caught by the new typing because `left` does not present in `ForStatement` node. Here so we can narrow `isFor` to `isForXStatement`.